yjit/src/cruby.rs

```diff
@@ -108,6 +108,20 @@ extern "C" {
     pub fn alloc_exec_mem(mem_size: u32) -> *mut u8;
 }
 
+#[cfg(not(test))]
```

Comment:
You can also put the `#[cfg(not(test))]` directly in the `insn_len()` function, next to the expression it applies to.

Reply:
True. Then I'd *also* probably want to turn off the "unreachable" warnings on the following code, or turn all the code on or off expression-by-expression. It winds up (to my eye) a lot messier than just having two single-expression functions.

Reply:
I think I'd prefer wrapping the two blocks inside the function because it avoids duplicating the signature.

```
pub fn insn_len(opcode:usize) -> u32
{
    #[cfg(test)]
    panic!("insn_len is a CRuby function, and we don't link against CRuby for Rust testing!");

    #[cfg(not(test))]
    unsafe {
        raw_insn_len(VALUE(opcode))
    }
}
```